src/Analyzer/Passes/IfTransformStringsToEnumPass.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <Analyzer/IQueryTreePass.h>
+
+namespace DB
+{
+
+/**
+ * This pass replaces string-type arguments in If and Transform to enum.
```

Comment:
Consider to add small example how query look like initially and how it look like after optimization.